casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -218,25 +221,66 @@ object BlockAPI {
     }
   }
 
-  def getBlocksResponse[F[_]: Monad: MultiParentCasperRef: Log: SafetyOracle: BlockStore]
-    : F[BlocksResponse] = {
+  def showBlocks[F[_]: Monad: MultiParentCasperRef: Log: SafetyOracle: BlockStore](
+      depth: Long
+  ): F[List[BlockInfosAtHeight]] = {
+    def casperResponse(implicit casper: MultiParentCasper[F]) =
+      for {
+        dag         <- MultiParentCasper[F].blockDag
+        startHeight = dag.topoSort.length - depth
```

Comment:
Something here is wrong. I think the goal of the PR is to only show blocks that are at most `depth` back from the current forkchoice, but this is not what is being done in this function. The height of the blocks at `dag.topoSort.head` is `dag.sortOffset`, so `startHeight` should be `dag.topoSort.length + dag.sortOffset - 1 - depth`. But more than that, the fold goes over the entire sort, so `depth` is actually not doing what it is supposed to. You should fold over `dag.topoSort.takeRight(depth)`.

Reply:
Nice catch; I'll make sure to include a test for this too.